Index: src/server/CServer_Game.cpp
===================================================================
--- src/server/CServer_Game.cpp	(revision 2855)
+++ src/server/CServer_Game.cpp	(working copy)
@@ -1066,6 +1066,7 @@
 
 			bool EndGame = false;
 
+			#if 0
 			//
 			// TEAM DEATHMATCH: Declare the team as winner
 			//
@@ -1315,7 +1316,9 @@
 			break; // TEAMDEATH
 
 			}
-
+			
+			#endif
+			
 			if( tLXOptions->tGameInfo.fTimeLimit > 0 && fServertime > tLXOptions->tGameInfo.fTimeLimit*60.0 ) {
 				if (networkTexts->sTimeLimit != "<none>")
 					SendGlobalText( networkTexts->sTimeLimit, TXT_NORMAL );
